### PR TITLE
fix(condo): DOMA-6345 AutoIncrementInteger field was improved

### PR DESCRIPTION
**Pull request by AleX83Xpert**
There is an ability to calculate the next value within dynamic scopes. For example: to create number within organization use the following parameters:
```
fields: {
    ...
    number: {
        schemaDoc: 'The news item number',
        type: AutoIncrementInteger,
        isRequired: false,
        autoIncrementScopeFields: ['organization'],
    },
    ...
}
```

### CODE REVIEW

File: apps/condo/domains/news/schema/NewsItem.test.js [hunk 1]
```diff
@@ -137,6 +138,21 @@ describe('NewsItems', () => {
                     await createTestNewsItem(anonymousClient, dummyO10n)
                 })
             })
+
+            test('The news item number autoincrement works fine', async () => {
+                const [o10n1] = await createTestOrganization(adminClient)
```
Comment (SavelevMatthew) [inline apps/condo/domains/news/schema/NewsItem.test.js:143]: o10n?)

File: packages/keystone/fields/AutoIncrementInteger/Implementation.js [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
-    
+
     async resolveInput ({ resolvedData, existingItem }) {
         if (!existingItem && !resolvedData[this.path]) {
-            const next = await this.adapter.nextValue()
+            const scopeFields = get(this, ['config', 'autoIncrementScopeFields'], [])
+            const scopeWhere = scopeFields.reduce((result, field) => {
+                const fieldValue = get(resolvedData, field)
+                if (fieldValue) {
+                    return { ...result, [field]: fieldValue }
```
Comment (SavelevMatthew) [inline packages/keystone/fields/AutoIncrementInteger/Implementation.js:15]: Copy inside reduce? How about assigning property to result and just return result itself?

File: packages/keystone/schema.js [hunk 3]
```diff
@@ -142,20 +142,6 @@ async function find (schemaName, condition) {
     return await schemaList._keystone.lists[schemaName].adapter.find(condition)
 }
 
-/**
- * @param {string} schemaName
- * @param {Object} condition
- * @returns {Promise<number>}
- */
-async function count (schemaName, condition) {
```
Comment (SavelevMatthew) [inline packages/keystone/schema.js:150]: why did you remove count?

Comment (SavelevMatthew) [reply packages/keystone/schema.js:150]: Seems not relative to this PR

File: apps/condo/domains/news/schema/NewsItem.test.js (hunk 1 above)
Comment (AleX83Xpert) [reply apps/condo/domains/news/schema/NewsItem.test.js:143]: o~~rganizatio~~n :)

File: packages/keystone/schema.js (hunk 3 above)
Comment (AleX83Xpert) [reply packages/keystone/schema.js:150]: I was asked about deleting of this unused function by @pahaz and decided to do it within this PR.

File: packages/keystone/fields/AutoIncrementInteger/Implementation.js (hunk 2 above)
Comment (AleX83Xpert) [reply packages/keystone/fields/AutoIncrementInteger/Implementation.js:15]: I'd like the reducer function to be pure, to make the code functional-ish. Yes, there are no disadvantages if I use mutating of the result object here. The `scopeFields` object always will be small, so there are no pros/cons for these two approaches for this task.

**Review by pahaz — APPROVED** [review]
